### PR TITLE
Complete u128 / safe.u128

### LINKED ISSUE #35: Implement 128-bit types necessary for NEAR blockchain runtime
Currently different types in the library are in various stages of implementation completeness.

NEAR uses 128-bit numbers internally, therefore we need following types fully implemented:
- [ ] `u128`
- [ ] `i128`
- [ ] `safe.u128`
- [ ] `safe.i128`

Acceptance criteria:
- [ ] All types listed above implement all operations currently supported by `u128`
- [ ] `safe.` types also implement overflow check for all operations which can result in overflow
- [ ] Code has 100% branch coverage in tests

### PR DESCRIPTION
**Pull request by MaxGraey**
Breaking changes:
- rename `u128.cmp` to `u128.ord`
- remove `u128#toF64Unsafe` method

Improvments:
- Add pre / post inc / dec operations and tests for `safe.u128`.
- Modernize code and tests
- Improve overflow detection for multiplication of `safe.u128`
- Improve inc / dec / neg routines for `u128`
- other size & perf improvements

Partially fix #35

### CODE REVIEW

**Comment by MaxGraey** [comment]
cc @willemneal @vgrichina

File: assembly/integer/safe/u128.ts [hunk 1]
```diff
@@ -101,19 +101,21 @@ export class u128 extends U128 {
 
   static fromBytesLE(array: u8[]): u128 {
     assert(array.length && (array.length & 15) == 0);
+    // @ts-ignore
     var buffer = array.dataStart;
     return new u128(
-      load<u64>(buffer, 0),
+      load<u64>(buffer, 0 * sizeof<u64>()),
```
Comment (willemneal) [inline assembly/integer/safe/u128.ts]: Is there a reason for this?  Won't this just get optimized away?

Comment (MaxGraey) [reply assembly/integer/safe/u128.ts]: Yes, it's just for consistency

Comment (willemneal) [reply assembly/integer/safe/u128.ts]: Okay

**Pull request merged**